### PR TITLE
chore: set up semantic-pull-request GitHub Action

### PR DESCRIPTION
**Pull request by flotwig**
- Closes 

### Additional details

- Switches from the semantic PR probot added in #7946 to the https://github.com/amannn/action-semantic-pull-request GitHub Action.
- Motivated by downtime on the probot, including the current multi-day outage
- Example of failure PR and output:
	- https://github.com/flotwig/cypress-workflow-test/pull/4
	- https://github.com/flotwig/cypress-workflow-test/runs/5112017104?check_suite_focus=true

### PR Tasks

- [na] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

Post-Merge:

- [ ] Disable Github App
- [ ] Add this check to required checks

### CODE REVIEW

File: .github/workflows/semantic-pull-request.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      - opened
+      - edited
+      - synchronize
+
+jobs:
+  main:
+    name: Lint Title
+    runs-on: ubuntu-latest
+    steps:
+      - uses: amannn/action-semantic-pull-request@v4
```
[the remark this answers is not in the document]
Comment (flotwig) [reply .github/workflows/semantic-pull-request.yml:15]: Doesn't seem to be. Should I fork it to `cypress-io` org to [lock it down](https://devopsjournal.io/blog/2021/02/06/GitHub-Actions-Forking-Repositories)?

[the remark this answers is not in the document]
Comment (flotwig) [reply .github/workflows/semantic-pull-request.yml]: Switched to using a fork under the `cypress-io` org.

File: .github/workflows/semantic-pull-request.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+name: "Semantic PR"
```
Comment (emilyrohrbough) [inline .github/workflows/semantic-pull-request.yml:1]: Lets match the naming that's been in use.
```suggestion
name: "Semantic Pull Request"
```

File: .github/workflows/semantic-pull-request.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (BlueWinds) [reply .github/workflows/semantic-pull-request.yml]: My only ask is that we add a comment here explaining why we're pointing to a forked version of the repo. And maybe in the forked version of the repo's README as well, though that's less important.

**Review by ryanthemanuel — APPROVED** [review]

File: .github/workflows/semantic-pull-request.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (emilyrohrbough) [reply .github/workflows/semantic-pull-request.yml]: Discussed during our sync today. Moving forward with forked repository. 👍🏻

**Review by emilyrohrbough — APPROVED** [review]

**Pull request merged**